### ISSUE TITLE
Not working properly with Ubuntu 20.10

### ISSUE THREAD

**Issue by AryaJohary** [slgobinath/SafeEyes#400]
Recently, my Ubuntu PC got an update to 20.10 version. I had Safe Eyes previously installed on 20.04 version where it worked just perfectly. But after update, it has got many issue. While taking breaks, it just gets hanged for a while and then resumes at about 3-2 remaining seconds. Each time a break is there, or if I try to take a break manually, same thing repeats.
Please take a look at its behaviour in Ubuntu 20.10 system. 
It is a very good app and I want to keep it using but after my system upgrade, it has became just annoying. 
Thanks very much.
Hope you will fix it soon.

**Comment by naggie** [comment]
I can confirm the same thing on Pop!_OS 20.10. It works, but causes windows to re-arrange and indeed hangs for a bit.

**Comment by PeterFajner** [comment]
I'm getting the same thing, my shell crashes whenever safeeyes takes a break. I'm on Pop!_OS 20.10, and the crashing started as soon as I upgraded from 20.04. This is my syslog during the crash:

```
Dec 12 14:27:48 localhost gnome-shell[27272]: JS ERROR: TypeError: actor.__animationInfo is undefined#012_sizeChangedWindow@resource:///org/gnome/shell/ui/windowManager.js:1385:9
Dec 12 14:27:48 localhost gnome-shell[27272]: JS ERROR: TypeError: actor.__animationInfo is undefined#012_sizeChangedWindow@resource:///org/gnome/shell/ui/windowManager.js:1385:9
Dec 12 14:27:48 localhost gnome-shell[27272]: Error in freeze/thaw accounting
Dec 12 14:27:48 localhost kernel: [ 5211.286279] traps: gnome-shell[27272] trap int3 ip:7fb6839e4517 sp:7ffdb735cbf0 error:0 in libglib-2.0.so.0.6600.1[7fb6839a6000+8b000]
Dec 12 14:27:48 localhost systemd[7188]: org.gnome.Shell@x11.service: Main process exited, code=dumped, status=5/TRAP
Dec 12 14:27:48 localhost systemd[7188]: org.gnome.Shell@x11.service: Failed with result 'core-dump'.
Dec 12 14:27:48 localhost systemd[7188]: org.gnome.Shell@x11.service: Scheduled restart job, restart counter is at 6.
Dec 12 14:27:48 localhost systemd[7188]: Stopped GNOME Shell on X11.
Dec 12 14:27:48 localhost systemd[7188]: Starting GNOME Shell on X11...
Dec 12 14:27:48 localhost /usr/libexec/gdm-x-session[7400]: (--) NVIDIA(GPU-0): BenQ GW2255 (DFP-0): connected
Dec 12 14:27:48 localhost /usr/libexec/gdm-x-session[7400]: (--) NVIDIA(GPU-0): BenQ GW2255 (DFP-0): Internal TMDS
Dec 12 14:27:48 localhost /usr/libexec/gdm-x-session[7400]: (--) NVIDIA(GPU-0): BenQ GW2255 (DFP-0): 330.0 MHz maximum pixel clock
Dec 12 14:27:48 localhost /usr/libexec/gdm-x-session[7400]: (--) NVIDIA(GPU-0):
Dec 12 14:27:48 localhost /usr/libexec/gdm-x-session[7400]: (--) NVIDIA(GPU-0): Ancor Communications Inc ASUS VC239 (DFP-1): connected
Dec 12 14:27:48 localhost /usr/libexec/gdm-x-session[7400]: (--) NVIDIA(GPU-0): Ancor Communications Inc ASUS VC239 (DFP-1): Internal TMDS
Dec 12 14:27:48 localhost /usr/libexec/gdm-x-session[7400]: (--) NVIDIA(GPU-0): Ancor Communications Inc ASUS VC239 (DFP-1): 600.0 MHz maximum pixel clock
Dec 12 14:27:48 localhost /usr/libexec/gdm-x-session[7400]: (--) NVIDIA(GPU-0):
Dec 12 14:27:48 localhost /usr/libexec/gdm-x-session[7400]: (--) NVIDIA(GPU-0): DFP-2: disconnected
Dec 12 14:27:48 localhost /usr/libexec/gdm-x-session[7400]: (--) NVIDIA(GPU-0): DFP-2: Internal DisplayPort
Dec 12 14:27:48 localhost /usr/libexec/gdm-x-session[7400]: (--) NVIDIA(GPU-0): DFP-2: 1440.0 MHz maximum pixel clock
Dec 12 14:27:48 localhost /usr/libexec/gdm-x-session[7400]: (--) NVIDIA(GPU-0):
Dec 12 14:27:48 localhost /usr/libexec/gdm-x-session[7400]: (--) NVIDIA(GPU-0): DFP-3: disconnected
Dec 12 14:27:48 localhost /usr/libexec/gdm-x-session[7400]: (--) NVIDIA(GPU-0): DFP-3: Internal TMDS
Dec 12 14:27:48 localhost /usr/libexec/gdm-x-session[7400]: (--) NVIDIA(GPU-0): DFP-3: 165.0 MHz maximum pixel clock
Dec 12 14:27:48 localhost /usr/libexec/gdm-x-session[7400]: (--) NVIDIA(GPU-0):
Dec 12 14:27:48 localhost gsd-media-keys[8579]: Failed to grab accelerators: GDBus.Error:org.freedesktop.DBus.Error.UnknownMethod: No such interface “org.gnome.Shell” on object at path /org/gnome/Shell
Dec 12 14:27:49 localhost gnome-shell[27672]: Some code accessed the property 'CredentialManager' on the module 'credentialManager'. That property was defined with 'let' or 'const' inside the module. This was previously supported, but is not correct according to the ES6 standard. Any symbols to be exported from a module must be defined with 'var'. The property access will work as previously for the time being, but please fix your code anyway.
Dec 12 14:27:49 localhost gnome-shell[27672]: Skipping parental controls support as it’s disabled
Dec 12 14:27:49 localhost gnome-shell[27672]: Unset XDG_SESSION_ID, getCurrentSessionProxy() called outside a user session. Asking logind directly.
Dec 12 14:27:49 localhost gnome-shell[27672]: Will monitor session 4
Dec 12 14:27:49 localhost dbus-daemon[7210]: [session uid=1000 pid=7210] Activating service name='org.freedesktop.portal.IBus' requested by ':1.270' (uid=1000 pid=27684 comm="ibus-daemon --panel disable --xim " label="unconfined")
Dec 12 14:27:49 localhost dbus-daemon[7210]: [session uid=1000 pid=7210] Successfully activated service 'org.freedesktop.portal.IBus'
Dec 12 14:27:49 localhost gnome-shell[27672]: Telepathy is not available, chat integration will be disabled.
Dec 12 14:27:49 localhost gsd-media-keys[8579]: Failed to grab accelerators: GDBus.Error:org.freedesktop.DBus.Error.UnknownMethod: No such interface “org.gnome.Shell” on object at path /org/gnome/Shell
Dec 12 14:27:49 localhost gnome-shell[27672]: Failed to create file /run/user/1000/gnome-shell-disable-extensions: Error opening file “/run/user/1000/gnome-shell-disable-extensions”: File exists
Dec 12 14:27:49 localhost gnome-shell[27672]: JS ERROR: Extension temperature@xtranophilist: SyntaxError: missing ( after for @ /home/peter/.local/share/gnome-shell/extensions/temperature@xtranophilist/extension.js:140
Dec 12 14:27:49 localhost gnome-shell[27672]: Failed to import DBusMenu, quicklists are not avaialble: Error: Requiring Dbusmenu, version none: Typelib file for namespace 'Dbusmenu' (any version) not found
Dec 12 14:27:49 localhost dbus-daemon[7210]: [session uid=1000 pid=7210] Activating service name='org.freedesktop.FileManager1' requested by ':1.269' (uid=1000 pid=27672 comm="/usr/bin/gnome-shell " label="unconfined")
Dec 12 14:27:49 localhost gnome-shell[27672]: Impossible to monitor trash: Gio.IOErrorEnum: Operation not supported
Dec 12 14:27:49 localhost gnome-shell[27672]: Impossible to enumerate trash children: Gio.IOErrorEnum: Operation not supported
Dec 12 14:27:50 localhost gnome-shell[27672]: g_file_enumerator_iterate: assertion 'direnum != NULL' failed
Dec 12 14:27:50 localhost gnome-shell[27672]: message repeated 6 times: [ g_file_enumerator_iterate: assertion 'direnum != NULL' failed]
Dec 12 14:27:50 localhost dbus-daemon[7210]: [session uid=1000 pid=7210] Successfully activated service 'org.freedesktop.FileManager1'
Dec 12 14:27:50 localhost system76-power[1012]: [INFO] DBUS Received GetSwitchable() method
Dec 12 14:27:50 localhost system76-power[1012]: [INFO] DBUS Received GetProfile() method
Dec 12 14:27:50 localhost gnome-shell[27672]: gnome-shell-extension-system76-power: power profile was set: 'Balanced'
Dec 12 14:27:50 localhost dbus-daemon[7210]: [session uid=1000 pid=7210] Activating service name='org.gnome.Shell.Notifications' requested by ':1.269' (uid=1000 pid=27672 comm="/usr/bin/gnome-shell " label="unconfined")
Dec 12 14:27:50 localhost dbus-daemon[7210]: [session uid=1000 pid=7210] Successfully activated service 'org.gnome.Shell.Notifications'
Dec 12 14:27:50 localhost systemd[7188]: Started GNOME Shell on X11.
Dec 12 14:27:50 localhost NetworkManager[985]: <info>  [1607797670.4347] agent-manager: agent[9d2423dc84550e4d,:1.263/org.gnome.Shell.NetworkAgent/1000]: agent registered
Dec 12 14:27:50 localhost gnome-shell[27672]: Error looking up permission: GDBus.Error:org.freedesktop.portal.Error.NotFound: No entry for geolocation
Dec 12 14:27:50 localhost gnome-shell[27672]: Window manager warning: Overwriting existing binding of keysym 71 with keysym 71 (keycode 18).
Dec 12 14:27:50 localhost gsd-media-keys[8579]: Failed to grab accelerator for keybinding settings:rfkill
Dec 12 14:27:50 localhost gsd-media-keys[8579]: Failed to grab accelerator for keybinding settings:playback-random
Dec 12 14:27:50 localhost gsd-media-keys[8579]: Failed to grab accelerator for keybinding settings:playback-repeat
Dec 12 14:27:50 localhost gsd-media-keys[8579]: Failed to grab accelerator for keybinding settings:rotate-video-lock
Dec 12 14:27:50 localhost gsd-media-keys[8579]: Failed to grab accelerator for keybinding settings:rotate-video-lock
Dec 12 14:27:50 localhost gsd-media-keys[8579]: Failed to grab accelerator for keybinding settings:hibernate
Dec 12 14:27:50 localhost gnome-shell[27672]: Bogus presentation time 0 travelled back in time, using current time.
Dec 12 14:27:51 localhost gnome-shell[27672]: JS ERROR: Error: No signal 'allocation-changed' on object 'StWidget'#012_init@/home/peter/.local/share/gnome-shell/extensions/desktop-icons@csoriano/desktopGrid.js:130:20#012_addDesktopIcons/<@/home/peter/.local/share/gnome-shell/extensions/desktop-icons@csoriano/desktopManager.js:193:27#012forEachBackgroundManager@/home/peter/.local/share/gnome-shell/extensions/desktop-icons@csoriano/desktopManager.js:766:36#012_addDesktopIcons@/home/peter/.local/share/gnome-shell/extensions/desktop-icons@csoriano/desktopManager.js:192:33#012_init@/home/peter/.local/share/gnome-shell/extensions/desktop-icons@csoriano/desktopManager.js:89:14#012innerEnable@/home/peter/.local/share/gnome-shell/extensions/desktop-icons@csoriano/extension.js:64:22#012enable/_startupPreparedId<@/home/peter/.local/share/gnome-shell/extensions/desktop-icons@csoriano/extension.js:53:94#012_startupAnimationComplete@resource:///org/gnome/shell/ui/layout.js:733:14#012onComplete@resource:///org/gnome/shell/ui/layout.js:711:36#012_makeEaseCallback/<@resource:///org/gnome/shell/ui/environment.js:84:13#012_easeActor/<@resource:///org/gnome/shell/ui/environment.js:165:64
Dec 12 14:27:51 localhost gnome-shell[27672]: GNOME Shell started at Sat Dec 12 2020 14:27:49 GMT-0400 (AST)
Dec 12 14:27:51 localhost gnome-shell[27672]: Registering session with GDM
```

**Comment by slgobinath** [comment]
Hi,
Thank you for taking your time to report the bug. Could you please check if this is related to the recently fixed bug https://github.com/slgobinath/SafeEyes/issues/392

I'd appreciate it if you can run from source code to test if the problem has been fixed:

```
git clone https://github.com/slgobinath/SafeEyes.git
cd SafeEyes

# To run from source, quit your running Safe Eyes instance and run the following command
python3 -m safeeyes
```

**Comment by PeterFajner** [comment]
Yup, running the version with that fix works perfectly for me!

**Comment by magikid** [comment]
I'm on Pop!_OS 20.10 and the version on master also resolved it for me.  My tiled windows all stayed exactly where they were.

**Comment by slgobinath** [comment]
Fixed version 2.1.0 has been released. It will be shortly available on the repo.
Merry Christmas!

**Comment by naggie** [comment]
Thanks @slgobinath, merry christmas to you too.
